tbdex/rfq/rfq.go

```diff
@@ -59,19 +65,86 @@ func (r *RFQ) UnmarshalJSON(data []byte) error {
 	// 	return errors.New("signer: %w does not match message metadata from: %w", decoded.Header.SignerDID, r.MessageMetadata.From)
 	// }
 
-	// TODO verify private data
-	// if requirePrivateData {
-	// 	err = r.verifyAllPrivateData()
-	// 	if err != nil {
-	// 	return fmt.Errorf("failed to verify private data: %w", err)
+	err = r.verifyPrivateData(privateDataStrict)
+	if err != nil {
+		return fmt.Errorf("failed to verify private data: %w", err)
+	}
 
-	// 	}
-	// }
+	return nil
+}
+
+// Parse validates, parses input data into an RFQ, and verifies the signature and private data.
+func (r *RFQ) Parse(data []byte, privateDataStrict bool) error {
+	err := r.UnmarshalJSON(data)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal RFQ: %w", err)
+	}
+
+	err = r.Verify(privateDataStrict)
+	if err != nil {
+		return fmt.Errorf("failed to verify RFQ: %w", err)
+	}
 
 	return nil
 }
 
-func (r *RFQ) verifyAllPrivateData() error {
+// Digest computes a hash of the rfq
+func (r RFQ) Digest() ([]byte, error) {
+	payload := map[string]any{"metadata": r.MessageMetadata, "data": r.Data}
+
+	hashed, err := tbdex.DigestJSON(payload)
+	if err != nil {
+		return nil, fmt.Errorf("failed to digest rfq: %w", err)
+	}
+
+	return hashed, nil
+}
+
+func (r *RFQ) verifyPrivateData(strict bool) error {
```

Comment:
seeking feedback on this function, prioritized collapsing both strict/not strict into a single method but could split into 2 if we think the readability outweighs

Reply:
3 levels of nested `if`s is kinda hard to read, but i won't make you split it into two methods if you prefer it this way. I preferred this pattern that I used in [tbdex-js](https://github.com/TBD54566975/tbdex-js/blob/main/packages/protocol/src/message-kinds/rfq.ts#L180-L256). Splitting out methods `verifyAllPrivateData` and `verifyPresentPrivateData` which maps roughly to `strict == true` and `strict == false`. Those two methods each call `verifyClaimsHash`, `verifyPayinHash` and `verifyPayoutHash`. This kept each method pretty focused. Ultimately it's a matter of style, so I'll leave it up to you which feels better.